### PR TITLE
50 free blueprints

### PR DESCRIPTION
**Pull request by Dimava**
Limit lowers amount of holded BP shapes to 50

### CODE REVIEW

**Comment by Dimava** [comment]
This requires level tweaking, level 6.5 was made purely for testing

**Comment by tobspr** [comment]
Uhm what exactly is the plan? :P

**Comment by Dimava** [comment]
When you unlock BPs you get 50 permanent shapes that will be restored immedately in case you spend them on making BP

File: src/js/game/hud/parts/blueprint_placer.js [hunk 1]
```diff
@@ -107,6 +107,11 @@ export class HUDBlueprintPlacer extends BaseHUDPart {
             const cost = blueprint.getCost();
             this.root.hubGoals.takeShapeByKey(blueprintShape, cost);
 
+            const leftover = this.root.hubGoals.getShapesStoredByKey(blueprintShape);
+            if (leftover < 50) {
+                this.root.hubGoals.putShapeByKey(blueprintShape, 50 - leftover);
+            }
+
```
Comment (Dimava) [inline src/js/game/hud/parts/blueprint_placer.js:114]: here it is

Comment (tobspr) [reply src/js/game/hud/parts/blueprint_placer.js:114]: So whenever you go below 50 shapes, you get it filled up again?

Comment (Dimava) [reply src/js/game/hud/parts/blueprint_placer.js:114]: yes

**Comment by tobspr** [comment]
I think this is pretty hard to explain to the user. I will try to find a different solution. Closing for now

**Pull request closed without merging**